client/securedrop_client/logic.py

```diff
@@ -981,8 +1005,21 @@ def on_file_download_failure(self, exception: Exception) -> None:
         else:
             if isinstance(exception, DownloadDecryptionException):
                 logger.error("Failed to decrypt %s", exception.uuid)
-                f = self.get_file(exception.uuid)
+                try:
+                    f = storage.get_file(self.session, exception.uuid)
+                except storage.SDDatabaseError as e:
+                    # This isn't necessarily an error; the file may have been deleted
+                    # at the server and has been removed from the database.
+                    # Log it, but don't panic
+                    logger.warning(
+                        "File download failure for uuid not in database (likely deleted record)."
+                    )
+                    logger.debug(f"File {exception.uuid} not found in database: {e}")
+                    return
+
                 self.file_missing.emit(f.source.uuid, f.uuid, str(f))
+            # TODO: This is not a great status message. There could be multiple downloads
```

Comment:
Please also file a bug for this :)

Reply:
https://github.com/freedomofpress/securedrop-client/issues/2265 :)